Design/active subscription

jduong09 opened the pull request:
### Feature
- Added 'active' css design to subscriptions that were in an active state.
- Removed active state when trying to create subscription.

ziggysauce commented on client/src/components/subscription/subscriptionsList.js:
```diff
@@ -1,12 +1,17 @@
+const SubscriptionsList = ({ subscriptions, setActiveSubscription, handleDashboard, handleDelete, openDeleteModal, activeSubscription }) => {
   const subscriptionsList = subscriptions.map((subscription) => {
     const { subscriptionUuid } = subscription;
+    let className="list-group-item p-0 mt-2 list-group-item-action border border-dark rounded-1";
+    if (activeSubscription.subscriptionUuid === subscriptionUuid) {
+      className += " bg-primary";
+    }
```
i've used [this library](https://www.npmjs.com/package/classnames) to better handle conditional classes but it's certainly not necessary. just something i want you to be aware of in case you wanted to look into it

jduong09 replied:
Thanks for bringing my attention to it, I'm good on this though.

ziggysauce commented on client/src/components/user/dashboard.js:
```diff
@@ -83,20 +82,22 @@ class Dashboard extends React.Component {
   handleDashboardChange(newView) {
     const { mainComponentView } = this.state;
-
+    
```
get rid of these spaces

ziggysauce commented on client/src/components/user/dashboard.js:
```diff
@@ -83,20 +82,22 @@ class Dashboard extends React.Component {
       return;
     }
 
+
```
get rid of this extra empty line

ziggysauce commented on client/src/components/user/dashboard.js:
```diff
@@ -83,20 +82,22 @@ class Dashboard extends React.Component {
+    
     if (mainComponentView === 'dashboardCalendar' || mainComponentView === 'subscriptionDetail') {
-      this.setState({ mainComponentView: newView });
+      const newState = (newView === 'createSubscription') ? { mainComponentView: newView, activeSubscription: false } : { mainComponentView: newView };
```
don't need parens here

```suggestion
      const newState = newView === 'createSubscription' ? { mainComponentView: newView, activeSubscription: false } : { mainComponentView: newView };
```

ziggysauce commented on client/src/components/user/dashboard.js:
```diff
@@ -68,7 +68,6 @@ class Dashboard extends React.Component {
       if (dueDate.lateDueDate) {
         toast.error(`Your ${name} subscription was due on ${new Date(dueDate.lateDueDate).toLocaleDateString()}`, {
           autoClose: false,
-          style: { backgroundColor: 'red', color: '#000000' }
```
thank you hahaha

ziggysauce commented on client/src/components/user/dashboard.js:
```diff
@@ -329,7 +331,7 @@ class Dashboard extends React.Component {
                   onClick={() => this.handleDashboardChange('createSubscription')}
                 >
-                  + Create
+                  +Create
```
i think we should keep the space

ziggysauce commented on client/src/components/user/dashboard.js:
```diff
@@ -341,7 +343,7 @@ class Dashboard extends React.Component {
                   onClick={() => this.setState({ addingSubscription: true })}
                 >
-                  + Create
+                  +Create
```
same comment here

ziggysauce reviewed (changes requested):
These being action buttons on existing subscriptions shouldn't have the dashed border. It makes sense for creating a sub since that doesn't exist yet, but for editing/deleting we should keep these normal buttons.

<img width="1054" alt="Screen Shot 2022-03-29 at 9 28 07 PM" src="https://user-images.githubusercontent.com/25065329/160732655-f5586202-6849-4264-8ea2-3f579d04b7c6.png">

The contrast of the black text on our purple primary color isn't great for readability. i suggest using a lighter purple for the active state

### current:
<img width="379" alt="Screen Shot 2022-03-29 at 9 32 12 PM" src="https://user-images.githubusercontent.com/25065329/160733096-0810cc30-d598-4799-a8d9-356d2517c711.png">

### lighter:
<img width="376" alt="Screen Shot 2022-03-29 at 9 33 10 PM" src="https://user-images.githubusercontent.com/25065329/160733098-4fd7958e-852b-4c52-ae41-875bf08fa182.png">

Couple things here:
(1) let's hide the scroll bar here. i think the following css should do it
```
-ms-overflow-style: none;
scrollbar-width: none;
```
(2) I wonder if it may look better to either lighten the text color of the labels or make them bold. I can be convinced that it needs neither...
<img width="1153" alt="Screen Shot 2022-03-29 at 9 36 49 PM" src="https://user-images.githubusercontent.com/25065329/160733553-dd6c016d-7e88-48c3-9ffa-3221f61b9620.png">

I don't think the profile pic needs a border. I also think it should have some more padding between the picture and the center text. programatically it has the same padding as the logo on the left, but since the logo has white space it looks like it's more spaced out and we may want to compensate on the right side to make it look more balanced
<img width="414" alt="Screen Shot 2022-03-29 at 9 44 16 PM" src="https://user-images.githubusercontent.com/25065329/160734164-8ae99e2b-4213-423a-b61e-cbb9b85cb6f9.png">

Is it possible to make the purple circles smaller so that they aren't as close to other circles? I think subtle is better with the calendar, no need to make GIANT circles. I also think it should be a lighter purple similar to the other comment I have about the active state on the subscription cards
<img width="339" alt="Screen Shot 2022-03-29 at 9 46 32 PM" src="https://user-images.githubusercontent.com/25065329/160734354-ab3f3f25-e84a-4905-a3b6-ce75a0a1ec61.png">

jduong09 commented:
### Changes/Additions
- Separated calendar days from each other with border-collapse: separate
- Changed distances between elements on navigation bar.
- Added font-weight to labels in subscription forms.
- Added font-weight to 'today' date and 'jump to today' element in calendar.
- Made background-color for selected day in calendar and active subscription in subscription list lighter.
- Reverted changes to 'edit' and 'delete' buttons.
- Removed scroll design on offcanvas (drawer) on mobile devices.

ziggysauce reviewed (changes requested):
Couple of things here:
- When I am editing one sub, but click another sub, it highlights the new sub but the content doesn't reflect that if i click the "stay on page" button
- On mobile, the view is a bit confusing as well since the highlighted sub is changing but the content to edit isn't
https://user-images.githubusercontent.com/25065329/162337144-973e980c-a29e-4d4f-a2b2-f2ae471900af.mov

ziggysauce commented on client/src/components/subscription/subscriptionsList.js:
```diff
@@ -1,12 +1,17 @@
+const SubscriptionsList = ({ subscriptions, setActiveSubscription, handleDashboard, handleDelete, openDeleteModal, activeSubscription }) => {
   const subscriptionsList = subscriptions.map((subscription) => {
     const { subscriptionUuid } = subscription;
+    let className="list-group-item p-0 mt-2 list-group-item-action border border-dark rounded-1";
```
```suggestion
    let className = "list-group-item p-0 mt-2 list-group-item-action border border-dark rounded-1";
```

ziggysauce commented on client/src/custom.scss:
```diff
@@ -2,6 +2,19 @@
+}
+
+.navbar-brand {
+  margin-right: 0 !important;
```
you can utilize bootstrap to do this instead with with `me-0` class

jduong09 commented:
### Changes/Fixes
- Fixed issue where update form would not display correct updated form between user clicks.
- Fixed mobile issue where update form content was not changing on user click.
- Added functionality on offcanvas hide.
- Removed 'data-bs' attributes on offcanvas, and added custom functionality to handle opening and closing offcanvas.

ziggysauce reviewed (commented):
changes look good to me. the only thing:

on mobile, when editing a subscription, if i close the drawer we no longer show the "stay" or "discard changes" modal. If that was intentional then this is fine and we can merge it in. otherwise if you still wanted that "confirm" modal it's missing and no longer working on mobile
